Trie/Trie/Trie/Trie.swift

```diff
@@ -62,11 +60,32 @@ class Trie {
     fileprivate let root: Node
     fileprivate var wordCount: Int
 
-    /// Creats an empty trie.
-    init() {
+    /// Creates an empty trie.
+    override init() {
```

Comment:
I asked around and came up with this conclusion - When subclassing, always call the `super` implementation. 

In this case, the compiler did not complain because of a very special circumstance where the superclass (`NSObject`) had only 1 initializer. In this case, Swift will automatically call the superclasses initializer for you. 

According to the person I was speaking to, there has been some compiler bugs in the past where inserting the explicit `super.init()` actually changes behaviour. 

In light of this, I think doing an explicit `super.init()` is the right way to go here.